File: leetcode/prob485.cpp

#include <iostream>
#include <algorithm>
#include <string>
#include <vector>
#include <map>
#include <utility>

#include <cctype>

using namespace std;
/*
* Name: 463. Island Perimeter
* WebSite: https://leetcode.com/problems/island-perimeter/
* Description: 
*
*/
class Solution {
public:
    /*
    * Runtime:    ms
    * Your runtime beats  % of cpp submissions. 
    */
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxCon = 0;
        int curCon = 0;
        for (int v : nums) {
            if (v == 0) {
                maxCon = std::max(maxCon, curCon);
                curCon = 0;
            } else {
                curCon += v;
            }
        }
        return std::max(maxCon, curCon);
    }   
};

int main() 
{
    Solution s;

    std::cout << "method 1: \n";

    std::cout << std::endl;
    return 0;
}                

File: leetCode_Python/prob485.py
#!/usr/bin/env python3
#-*- coding:utf-8 -*-

# author    : lambda RDG
# Date&Time : 2019/03/30 11:27:45

"模块的文档注释"

# Name: 485. Max Consecutive Ones 
# Website: https://leetcode.com/problems/max-consecutive-ones/ 
# Description:
import sys

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maxCon = 0
        curCon = 0
        for v in nums:
            if v == 0:
                maxCon = max(maxCon, curCon)
                curCon = 0
                continue
            curCon += v
        maxCon = max(maxCon, curCon)
        return maxCon

if __name__ == "__main__":
    pass